src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -180,8 +179,29 @@ void CALLBACK cfApiFetchDataCallback(const CF_CALLBACK_INFO *callbackInfo, const
     }
 }
 
+void CALLBACK cfApiCancelFetchData(const CF_CALLBACK_INFO *callbackInfo, const CF_CALLBACK_PARAMETERS * /*callbackParameters*/)
+{
+    const auto path = QString(QString::fromWCharArray(callbackInfo->VolumeDosName) + QString::fromWCharArray(callbackInfo->NormalizedPath));
+
+    qInfo(lcCfApiWrapper) << "Cancel fetch data of" << path;
+
+    auto vfs = reinterpret_cast<OCC::VfsCfApi *>(callbackInfo->CallbackContext);
+    Q_ASSERT(vfs->metaObject()->className() == QByteArrayLiteral("OCC::VfsCfApi"));
+    const auto requestId = QString::number(callbackInfo->TransferKey.QuadPart, 16);
+    const auto cancelHydration = [=] {
+        vfs->cancelHydration(requestId, path);
+    };
+
+    const auto invokeResult = QMetaObject::invokeMethod(vfs, cancelHydration, Qt::QueuedConnection);
```

Comment:
style nitpick: I'd inline cancelHydration here, the intermediate variable is pretty much useless here. I find this more directly intelligible:
`const auto invokeResult = QMetaObject::invokeMethod(vfs, [=]{ vfs->cancelHydration(requestId, path); }, Qt::QueuedConnection);`